### PR TITLE
run team auditor on slow team loads

### PR DESCRIPTION
**Pull request by maxtaco**
- the team auditor more or less drops into place, but the one tricky aspect here is that we need to plumb through the "merkle head seqno", which is the last known merkle seqno when the team head was fabricated.
- we don't want to blow away loader caches, so if it's not there in the state, we backfill it, and check server replies against the previously verified chain link ID for seqno=1
- on a fresh load, we just load it from disk
- test that makes sure the back-fill of merkle head works

### CODE REVIEW

**Comment by maxtaco** [comment]
@mlsteele this was a pretty easy integration and should hopefully be an easy review!

File: go/teams/audit.go [hunk 1]
```diff
@@ -41,11 +41,23 @@ var mobileParams = AuditParams{
 	LRUSize:               500,
 }
 
+var devParams = AuditParams{
```
Comment (maxtaco) [inline go/teams/audit.go]: Don't want to slow down CI anymore.

File: go/teams/loader.go [hunk 2]
```diff
@@ -1402,3 +1408,61 @@ func (l *TeamLoader) NotifyTeamRename(ctx context.Context, id keybase1.TeamID, n
...
+	}
+	if !headLink.LinkID().Eq(expectedLink) {
+		return ret, NewInvalidLink(headLink, "wrong head link hash: %s != %s", headLink.LinkID, expectedLink)
+	}
+	if headLink.isStubbed() {
+		return ret, NewInvalidLink(headLink, "got a stubbed head link, but wasn't expecting that")
+	}
+	headMerkle := headLink.inner.Body.MerkleRoot.ToMerkleRootV2()
+	state.HeadMerkle = &headMerkle
+	return headMerkle.Seqno, nil
```
Comment (mlsteele) [inline go/teams/loader.go]: Does anything here check that headLink.inner matches headLink.outer?

Comment (maxtaco) [reply go/teams/loader.go]: ah good point, i just assumed the unpack function did that, but maybe that was a bad assumption

Comment (maxtaco) [reply go/teams/loader.go]: Great catch on that one!

**Comment by maxtaco** [comment]
thanks for the review!

**Pull request merged**